generators/aws-containers/aws-client.js

```diff
@@ -0,0 +1,299 @@
+/* global process */
+const _ = require('lodash');
+const fs = require('fs');
+const chalk = require('chalk');
+const shelljs = require('shelljs');
+
+const utils = require('../utils');
+
+const AwsSSM = require('./lib/ssm');
+const AwsECR = require('./lib/ecr');
+const AwsCF = require('./lib/cf');
+
+const DEFAULT_REGION = 'us-east-1';
+const S3_MIN_PART_SIZE = 5242880;
+
+
+// Instance from aws-sdk
+let AWS;
+let credentials;
+let ec2;
+// let ecr;
+let s3;
+let sts;
+let ora;
+
+// Instances from ./lib. Composed with aws-sdk
+let SSM;
+let ECR;
+let CF;
+
+let ProgressBar;
+
+module.exports = {
+    DEFAULT_REGION,
+    SSM: () => SSM,
+    ECR: () => ECR,
+    CF: () => CF,
+    createS3Bucket,
+    getDockerLogin,
+    listRegions,
+    listSubnets,
+    listVpcs,
+    loadAWS,
+    saveCredentialsInAWS,
+    initAwsStuff,
+    sanitizeBucketName,
+    uploadTemplate,
+};
+
+/**
+ * Will load the aws-sdk npm dependency if it's not already loaded.
+ *
+ * @param generator the yeoman generator it'll be loaded in.
+ * @returns {Promise} The promise will succeed if the aws-sdk has been loaded and fails if it couldn't be installed.
+ */
+function loadAWS(generator) {
+    return new Promise((resolve, reject) => {
+        try {
+            AWS = require('aws-sdk') // eslint-disable-line
+            ProgressBar = require('progress'); // eslint-disable-line
+            ora = require('ora'); // eslint-disable-line
+        } catch (e) {
+            generator.log('Installing AWS dependencies');
+            let installCommand = 'yarn add aws-sdk@^2.167.0 progress@2.0.0 ora@1.3.0';
+            if (generator.config.get('clientPackageManager') === 'npm') {
+                installCommand = 'npm install aws-sdk@^2.167.0 progress@2.0.0 ora@1.3.0--save';
```

Comment:
If the version changes we'd have to update it here too... is there a better solution?

Reply:
We'll definitely look into it but the issue with not specifying the version is that we don't have any control over the installed version. The code works with `2.167.0` so we're sure to avoid any breaking changes.

For example if `aws-sdk` has a 3.0.0 major release it might break the code. Hardcoding a version means that we only have to be careful at the npm version required by the `jhipster-gernerator` and update accordinlgy, which will happens less often than a `aws-sdk`update